src/averageAges.js

```diff
@@ -37,7 +44,23 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let women = people.filter(person => person.sex === 'f');
+
+  if (withChildren) {
+    women = women.filter(
+      woman => people.some(
+        person => person.mother === woman.name
+      )
```

Comment:
If you need to do something like this it's better to make a little different formatting
```
  if (withChildren) {
    women = women
      .filter(woman => people
        .some(person => person.mother === woman.name));
  }
```

Other thing you can find in checklist, paragraph 4